inspirehep/modules/records/tasks.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Celery tasks provided by Record module."""
+
+from __future__ import absolute_import, division, print_function
+
+from celery import current_app, shared_task
+from celery.utils.log import get_task_logger
+from elasticsearch.helpers import scan
+
+from invenio_search import current_search_client as es
+from invenio_db import db
+from invenio_records.api import Record
+
+from inspire_schemas import api
+from inspirehep.modules.search.utils import dotter
+
+logger = get_task_logger(__name__)
+
+
+@shared_task(ignore_result=True)
+def replace_old_ref_for_record(old_record_ref, new_record_ref):
+    possible_fields = get_ref_fields()
+    old_control_number = get_recid_from_url(old_record_ref)
+    record_uuids = es_search_pointing_ids_from_ref_in_terms(possible_fields, old_control_number)
+
+    for rec_uuid in record_uuids:
+        rec_db = Record.get_record(rec_uuid)
+        find_and_replace_old_link_with_new_link_in_record(rec_db, new_record_ref, old_record_ref)
+        rec_db.commit()
+    db.session.commit()
+
+
+def remove_dot(string):
+    return string[1:]
+
+
+def get_recid_from_url(url):
+    try:
+        res = int(url.split('/')[-1])
+    except ValueError:
+        res = None
+    return res
+
+
+def find_ref_fields_in_schema(collection, sum_fields):
+    bad_fields = ['deleted_recid', 'new_recid']
```

Comment:
Yeah, you can also read it as a global form inspirehep.config. In this way we can amend it in case of the schema evolving. Also I wouldn't call it `bad_fields` :) It's more `ignore_refs_in_fields`